force-app/main/default/lwc/barcodeScannerExample/__tests__/barcodeScannerExample.test.js

```diff
@@ -0,0 +1,95 @@
+import { createElement } from 'lwc';
+import { getNavigateCalledWith } from 'lightning/navigation';
+import BarcodeScannerExample from 'c/barcodeScannerExample';
+import { setBarcodeScannerAvailable } from 'lightning/mobileCapabilities';
+
+describe('c-barcode-scanner-example', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    // Helper function to wait until the microtask queue is empty.
+    // Used when having to wait for asynchronous DOM updates.
+    async function flushPromises() {
+        return Promise.resolve();
+    }
+
+    it('directs the user to the mobile app when Barcode Scanner is unavailable', async () => {
+        // Create initial BarcodeScannerExample element and attach to virtual DOM
+        const elementBarcodeScannerExample = createElement(
+            'c-barcode-scanner-example',
+            {
+                is: BarcodeScannerExample
+            }
+        );
+        document.body.appendChild(elementBarcodeScannerExample);
+
+        // Wait for element to mount
+        await flushPromises();
+
+        // Mount `Scan QR Code` button and trigger scan of property record ID
+        const elementScannerDirections =
+            elementBarcodeScannerExample.shadowRoot.querySelector(
+                // elementBarcodeScannerExample.template.querySelector(  // TODO: This was in the linked example but doesn't work?
```

Comment:
The problem is a bit more complex: in Jest, you can't use `innerText` because you're not in a real browser (see this [great answer](https://stackoverflow.com/questions/47902335/innertext-is-undefined-in-jest-test)).

As general guidance, you shouldn't test for the exact text in this case because it's hard coded in the markup. You would only check for text if it was dynamic.

What's more important to check in you case is that you can find the div itself.
Simply change your expect to this and the test will pass:
```js
expect(elementScannerDirections).not.toBeNull();
```

Reply:
Thank you for explaining, that makes much more sense! Checking the text seemed smelly to me as well 🤧